### ISSUE TITLE
Allow for multiple examples per type

### ISSUE THREAD

**Issue by ncoe** [stoicflame/enunciate#729]
I have started playing around with using enunciate to replace our SDK documentation.
The page that is generated for a type includes one and only one example based on the properties that it exposes.

1. What would be involved in including multiple examples?
2. What would be involved in either referencing a file with an example, or including an example inline.

Where this would help is that we support multiple media types, where one of the types has a special method for serializing the data. Having multiple examples would let us show the example even though enunciate does not know what that format looks like.

The second might be resolvable by #681.

**Comment by stoicflame** [comment]
See [`@responseExample`](https://github.com/stoicflame/enunciate/wiki/Enunciate-Specific-Annotations#responseexample-javadoc-tag), which allows referencing a file and example per response type.

**Comment by ncoe** [comment]
It looks like `@responseExample` is used for endpoints that can be accessed. For everything else, like an object used as a field in an object, the example is not even considered.

**Comment by stoicflame** [comment]
I'm trying to understand what the documentation would look like.

Let's say you have a data type described by example E<sub>a</sub> within context C<sub>a</sub>, where a "context" is a reference to the data type from another data type, a resource method, etc. The same data type is described by example E<sub>b</sub> within context C<sub>b</sub>.

Are you thinking both E<sub>a</sub> and E<sub>b</sub> would be listed on the documentation page for that data type? If so, wouldn't it be confusing to see E<sub>b</sub> from C<sub>a</sub> and vice-versa?

Of are you thinking that you'd like two documentation pages for the data type, each with a different example? If so, how would Enunciate be made context-aware? How would you associate E<sub>a</sub> with the field or method or class that defines C<sub>a</sub>? Configuration file? Some kind of annotation?

Note: a related concept is Enunciate's support for generics as specified at #400. It doesn't really account for multiple examples per type, but it does kind of capture the notion of a context.

**Comment by ncoe** [comment]
In one example, I have an object that essentially has a type field, and then additional fields to provide metadata. What I would like to do in this case is to provide examples for each type with only the other fields that are needed for metadata, as that is also how it gets serialized.

In another example, I have an object that has a field with a map datatype. We have a vendor specific media type that clients use to flatten how the data is presented to reduce the nesting required to get at data. I want to show what the object would look like for these different media types on their own pages in addition to contextual examples for the endpoints.

These come up because we have a smallish set of object that the endpoints consume and return to the client, that auto generating is not rich enough to articulate without assistance. The common theme is basically: what does a list look like and what does a map look like.

**Comment by stoicflame** [comment]
So wouldn't this case be better suited to user-written documentation in the JavaDoc of the data type or resource method? If the Enunciate-generated examples are inadequate, you could just disable them.

If that doesn't work, what kind of mechanism are you envisioning to provide Enunciate with the information needed to provide multiple documentation examples? And what would the generated HTML pages look like?

**Comment by ncoe** [comment]
You are correct that i could just disable all examples and use javadoc. What I wanted to replicate was the layout with the examples after the documentation of headers, parameters and media types like what is shown with auto-generated examples.
